lib/Utilities/FSAccess/FSAccessFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * @author Juan Pablo Villafáñez <jvillafanez@solidgeargroup.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license GPL-2.0
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+namespace OCA\DataExporter\Utilities\FSAccess;
+
+class FSAccessFactory {
```

Comment:
I tend to create factories only if we need different constructions, don't see why we can't construct directly, but :ok:

Reply:
I think more than 90% of the services registered in the DIContainer behave like a singleton class: although the class isn't designed as singleton, normally all the objects receive the same instance.

It's unlikely that we want the same instance in any place, so this means that we either define each service with the specific FSAccess instance to be used (increasing the boilerplate per class requiring this) or we define only one fixed instance to be used as service.
Note that the FSAccess instance is designed in a way that prevents changing the base root folder.

In addition, all the dependency wiring is defined before the commands are executed. The problem is that the target path where the exported info resides is known during the command execution. We can't place the jail during the dependency wiring because at that point is unknown.